ocaml/parsing/jane_syntax.ml

```diff
@@ -655,12 +647,23 @@ module Layouts = struct
         | _ -> raise Unexpected
       with
         Unexpected -> Desugaring_error.raise ~loc (Not_a_layout payload)
+
+    let bound_vars_from_vars_and_payload ~loc var_names payload =
+      let layouts = option_list_from_payload ~loc payload in
+      try
+        List.combine var_names layouts
+      with
+      (* seems silly to check the length in advance when [combine] does *)
+        Invalid_argument _ ->
+        Desugaring_error.raise ~loc
+          (Wrong_number_of_layouts(List.length var_names, layouts))
   end
 
   (*******************************************************)
-  (* Encoding *)
+  (* Encoding types *)
 
   let type_of ~loc ~attrs typ =
+    let module Ast_of = Ast_of (Core_type) (Ext) in
```

Comment:
Won't this recreate the module more times than necessary?  I'd hoist it back to the top level

Reply:
This is necessary, because it's specialized to the syntactic category. On the other hand, `Of_ast` is not similarly specialized, and so I hoisted that one.